File: problem169.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int major = nums[0];
        int count = 0;
        foreach (int num in nums) {
            if (count == 0) {
                major = num;
                count = 1;
            } else if (num == major) {
                count++;
            } else {
                count--;
            }
        }
        return major;
    }
}

File: su19/problem169.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currLargest = nums[0]
        count = 1
        i = 1
        while i < len(nums):
            if nums[i] == currLargest:
                count += 1
            elif count > 0:
                count -= 1
            else:
                currLargest = nums[i]
                count = 1
            i += 1
        return currLargest

        